src/control/lib/control/server_meta.go

```diff
@@ -37,26 +42,34 @@ type (
 		Pools   SmdPoolMap           `json:"pools"`
 	}
 
-	// SmdQueryReq contains the request parameters for a SMD query
-	// operation.
+	// SmdQueryReq contains the request parameters for a SMD query operation.
 	SmdQueryReq struct {
 		unaryRequest
-		OmitDevices      bool                 `json:"omit_devices"`
-		OmitPools        bool                 `json:"omit_pools"`
-		IncludeBioHealth bool                 `json:"include_bio_health"`
-		SetFaulty        bool                 `json:"set_faulty"`
-		UUID             string               `json:"uuid"` // UUID of pool or device for single result
-		Rank             ranklist.Rank        `json:"rank"`
-		Target           string               `json:"target"`
-		ReplaceUUID      string               `json:"replace_uuid"` // UUID of new device to replace storage
-		NoReint          bool                 `json:"no_reint"`     // for device replacement
-		Identify         bool                 `json:"identify"`     // for VMD LED device identification
-		StateMask        storage.NvmeDevState `json:"state_mask"`   // show devices with state matching mask
-	}
-
-	// SmdQueryResp represents the results of performing
-	// SMD query operations across a set of hosts.
-	SmdQueryResp struct {
+		OmitDevices      bool          `json:"omit_devices"`
+		OmitPools        bool          `json:"omit_pools"`
+		IncludeBioHealth bool          `json:"include_bio_health"`
+		UUID             string        `json:"uuid"`
+		Rank             ranklist.Rank `json:"rank"`
+		Target           string        `json:"target"`
+		FaultyDevsOnly   bool          `json:"-"` // only show faulty devices
+	}
+
+	// SmdManageReq contains the request parameters for a SMD query operation.
+	SmdManageReq struct {
+		unaryRequest
+		SetFaulty   bool          `json:"set_faulty"`
+		IDs         string        `json:"uuid"` // comma separated list of IDs
```

Comment:
From an API perspective, why not make this a slice? These are device UUIDs, right? Maybe UUIDs would be the best name in that case.

Reply:
At this stage this field is expected to carry either a single UUID for set-faulty or replace manage commands, but a list of UUIDs and/or PCI addresses for the LED variants. I can update the JSON tag from uuid->ids in the follow on PR but I think it's actually not used. The field could be changed to a slice but I don't see a strong reason to given the positional argument will be comma separated and we want to keep the unpacking and validation of tokens within the control API layer.

Reply:
Fair enough.

Reply:
updated the JSON tag name in PR https://github.com/daos-stack/daos/pull/10768